arch/arm/core/aarch64/mmu/arm_mmu.c

```diff
@@ -12,14 +12,16 @@
 #include <arch/arm/aarch64/arm_mmu.h>
 #include <linker/linker-defs.h>
 #include <sys/util.h>
-#include "arm_mmu.h"
 
-static uint64_t base_xlat_table[NUM_BASE_LEVEL_ENTRIES]
-		__aligned(NUM_BASE_LEVEL_ENTRIES * sizeof(uint64_t));
+#include "arm_mmu.h"
 
-static uint64_t xlat_tables[CONFIG_MAX_XLAT_TABLES][XLAT_TABLE_ENTRIES]
+static uint64_t kernel_xlat_tables[CONFIG_MAX_XLAT_TABLES * XLAT_TABLE_ENTRIES]
 		__aligned(XLAT_TABLE_ENTRIES * sizeof(uint64_t));
```

Comment:
There is a reason why `base_xlat_table` is kept separate. Generally, the `base_xlat_table` will have many limited number of entries - in a single digit - as decided by `NUM_BASE_LEVEL_ENTRIES`, so it does not make sense to allocate 4 kB for the same. I think same can still be maintained by having separate `base_xlat_table` entry in `struct arm_mmu_ptables`.

Reply:
ok , I see in the commit message that you are aware of this. "This is probably going to waste a bit of space"

Reply:
Yes, I know. The problem is that in the future we are probably going to create at compile time the set of kernel page tables and save that directly into some memory region (like it is done now for X86 by the `gen_mmu.py` script) and having one single region already in the code can help us a bit in that regard.